### PR TITLE
cat anomaly detectors specification

### PR DESCRIPTION
**Pull request by lcawl**
Adds missing fields for the [cat anomaly detectors API](https://www.elastic.co/guide/en/elasticsearch/reference/master/cat-anomaly-detectors.html). Also some minor fixes in the style guide.

### CODE REVIEW

**Comment by lcawl** [comment]
@delvedor the `make contrib` for this PR currently generates the following errors:

> ../specification/cat/_types/CatBase.ts
[error] ../specification/cat/_types/CatBase.ts: SyntaxError: An enum member name must be followed by a ',', '=', or '}'. (36:10)
[error]   34 |   assignment_explanation = 0,
[error]   35 |   /** @aliases bc, bucketsCount */
[error] > 36 |   buckets.count = 1,
[error]      |          ^
[error]   37 |   /** @aliases btea, bucketsTimeExpAvg */
[error]   38 |   buckets.time.exp_avg = 2
[error]   39 |   /** @aliases bteah, bucketsTimeExpAvgHour */

Is there some sort of escape character for the period in these values?

File: specification/cat/_types/CatBase.ts [hunk 1]
```diff
@@ -28,3 +28,126 @@ import { RequestBase } from '@_types/Base'
...
+  /** @aliases ni, nodeId */
+  node.id = 56
+  /** @aliases nn, nodeName */
+  node.name = 57
+  /** @aliases ot */
+  opened_time = 58
+  /** @aliases s */
+  state = 59
+}
+export type CatAnonalyDetectorColumns = CatAnonalyDetectorColumn | CatAnonalyDetectorColumns[]
```
Comment (lcawl) [inline specification/cat/_types/CatBase.ts:153]:
```suggestion
export type CatAnonalyDetectorColumns = CatAnonalyDetectorColumn | CatAnonalyDetectorColumn[]
```

File: specification/cat/ml_jobs/CatJobsRequest.ts [hunk 2]
```diff
@@ -56,9 +56,30 @@ export interface Request extends CatRequestBase {
...
+     * @server_default buckets.count,data.processed_records,forecasts.total,id,model.bytes,model.memory_status,state
+     */
+    h?: CatAnonalyDetectorColumns
+    /**
+     * If true, the response includes help information.
+     * @server_default false
+     */
+    help?: boolean
+    /** Comma-separated list of column names or column aliases used to sort the response. */
+    s?: string
```
Comment (lcawl) [inline specification/cat/ml_jobs/CatJobsRequest.ts:74]:
```suggestion
    s?: CatAnonalyDetectorColumns
```

File: specification/cat/_types/CatBase.ts [hunk 3]
```diff
@@ -28,3 +28,126 @@ import { RequestBase } from '@_types/Base'
 export class CatRequestBase
   extends RequestBase
   implements CommonCatQueryParameters {}
+
+export enum CatAnonalyDetectorColumn {
+  /** @aliases ae */
+  assignment_explanation = 0,
+  /** @aliases bc, bucketsCount */
+  buckets.count = 1,
```
Comment (delvedor) [inline specification/cat/_types/CatBase.ts:36]: You can't use a dot without wrapping the name inside quotes.
```suggestion
  'buckets.count' = 1,
```

Comment (lcawl) [reply specification/cat/_types/CatBase.ts]: Thanks!

**Pull request merged**